sdk/identity/azure-identity/src/main/java/com/azure/identity/ChainedTokenCredential.java

```diff
@@ -36,29 +38,20 @@
 
     @Override
     public Mono<AccessToken> getToken(TokenRequestContext request) {
-        List<CredentialUnavailableException> exceptions = new ArrayList<>(4);
+        StringBuilder message = new StringBuilder();
         return Flux.fromIterable(credentials)
-                   .flatMap(p -> p.getToken(request).onErrorResume(CredentialUnavailableException.class, t -> {
-                       exceptions.add(t);
+                   .flatMap(p -> p.getToken(request).onErrorResume(Exception.class, t -> {
+                       if (!t.getClass().getSimpleName().equals("CredentialUnavailableException")) {
+                           throw logger.logExceptionAsError(new CredentialUnavailableException(
```

Comment:
What will 'throw' achieve here ?

Reply:
In the reactive spec, this should return normally rather than throwing (ie. `return Mono.error(logger.logExceptionAsError)`).
https://github.com/reactive-streams/reactive-streams-jvm#2.13